### ISSUE TITLE
there is no spans for queries w/out exec() for mongoose 5.6.3 

### ISSUE THREAD

**Issue by evgeni-k** [elastic/apm-agent-nodejs#1550]
**Describe the bug**
The library does not automatically creates span if there is no explicit `exec()` function call on query object.

Mongoose uses native Promise.
```javascript
mongoose.Promise = global.Promise;
```

Following code snippet is instrumented perfectly and span is created
```javascript
async function test() {
    const result = await Model.find().exec();
    console.log(result);
}
```

This code snippet cannot be automatically instrumented (span cannot be created)
```javascript
async function test() {
    const result = await Model.find();
    console.log(result);
}
```

Mongoose automatically calls `exec` when `then` is called https://github.com/Automattic/mongoose/blob/5.6.3/lib/query.js#L4297-L4299 

**To Reproduce**

Run code snippets below in your app.
If you run with explicit `exec` call then span is created otherwise it won't.

**Expected behavior**

Span should be created w/ and w/out explicit `exec` function call.

**Environment (please complete the following information)**

- OS: Mac OS 10.15.1
- Node.js version: 12.13.0 
- APM Server version: 7.4.0
- Agent version: 3.1.0

**How are you starting the agent? (please tick one of the boxes)**

- [x] Calling `agent.start()` directly (e.g. `require('elastic-apm-node').start(...)`)
- [ ] Requiring `elastic-apm-node/start` from within the source code
- [ ] Starting node with `-r elastic-apm-node/start`

**Comment by Qard** [comment]
This is an unresolved issue with Promise Hooks in V8 and async_hooks in Node.js core. https://github.com/nodejs/node/issues/22360

The only way to make this work is to fix it in V8.

**Comment by evgeni-k** [comment]
Thank you for your reply. Probably it makes sense to update docs to explicitly highlight that fact about custom thenable. 

Feel free to close the ticket.

**Comment by sibelius** [comment]
does this also affects `.aggregate`?

**Comment by Systerr** [comment]
https://github.com/nodejs/node/issues/22360#issuecomment-653199762  async_hooks was fixed

**Comment by Qard** [comment]
Yep, I fixed the issue in V8 a few months ago. It's in Node.js 14.x. Still waiting on the backport to land in 12.x.

**Comment by david-luna** [comment]
Just came across this issue and did some test to see if still applies. Mongoose is added to the support list in `elastic-apm-agent@4`

I've used [this example](https://github.com/elastic/apm-agent-nodejs/blob/main/examples/trace-mongoose.js) from the main branch to test also with the latest version of the `3.x` branch.

```
req.waterfall:
    trace 9fe140
    `- transaction ed477c "t1" (34.813ms, outcome=unknown)
       `- span 8fac72 "example-trace-mongoose.people.insert" (1ms, mongodb)
       `- span bc7a36 "example-trace-mongoose.people.find" (1ms, mongodb)
       `- span b84aca "example-trace-mongoose.people.delete" (1ms, mongodb)
       `- span ec1f92 "admin.$cmd.endSessions" (0ms, mongodb, sync)
```
Mongoose eersions tested were: 7.6.3, 6.12.2, 5.13.21 & 5.6.3

So I'm inclined to close this one @evgeni-k if there is nothing else to look

**Comment by sibelius** [comment]
this is breaking for 

mongoose 8.03
mongodb 6.2.0

**Comment by sibelius** [comment]
It is working fine on latest version of packages
